subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -33,53 +33,60 @@ class DeviceWorker(
     fun run() = scope.launch {
 
         var state: State = try {
-            checkDeviceAlive()
-            device.state()
-        } catch (t: Exception) {
-            device.warn("Error while checking device initial state", t)
-            // No intention is lost. DeviceWorkerMessage.WorkerFailed event is unnecessary.
-            // Can't use this device any more. ReservationClient will get a new one.
-            return@launch
+            when (val status = device.deviceStatus()) {
+                is Device.DeviceStatus.Alive -> device.state()
+                is Device.DeviceStatus.Freeze -> {
+                // No intention is lost. DeviceWorkerMessage.WorkerFailed event is unnecessary.
+                // Can't use this device any more. TODO Will ReservationClient get a new one?
+                    device.warn("Device wasn't booted. Failed on initial run", status.reason)
+                    return@launch
+                }
+            }
+        } catch (t: Throwable) {
+            throw RuntimeException("Unexpected error when initialize a $device", t)
```

Comment:
It will kill the whole InstrumentationTask
Now there is no situation that can throw an exception from `device.deviceStatus()`

Reply:
Do we have any examples of such unexpected errors?

Reply:
Now, but it's a way I consider writing code in the offensive style